Added GPU Temperature and utilization

JeremiahPetersen opened the pull request:
This pull request introduces the following changes:

1. AMD Support: Implemented a check for the availability of the pyadl library and the accessibility of AMD drivers. This ensures that the application only attempts to retrieve AMD GPU details if they are accessible.
2. Temperature Information: A new section, "Temperature," has been added to the GUI to display the temperature details of the CPU and GPUs (both NVIDIA and AMD). It uses psutil, GPUtil, pyadl, and wmi to fetch the temperature details based on the system and platform.
3. Error Handling in Graphics Section: Enhanced the error handling in the graphics section to display a message if there's an issue fetching the AMD GPU details.

Please review and let me know of any feedback or changes required.

JeremiahPetersen committed:
added cpu and gpu temp

JeremiahPetersen committed:
gpu temp 1 sec updates added

diamant3 commented:
Thank you so much for your PR!

By the way, I based the SYSIG app into a major parts of a computer, please don't separate the section for minor section like temperature, I suggest to add that temp info to processor section and Graphics section and please minimize the use of modules as possible and lastly, run `pip install -r requirements.txt` first for every OS. 

Sorry for my long and demanding request because I don't have any build workflow(just a linter 😢) for every possible OS to test.

JeremiahPetersen commented:
no worries at all, thank you for the feedback!  I will work on these adjustments in the next 24 hours :)

diamant3 commented:
Okay I will review that asap, thanks again!

JeremiahPetersen committed:
moved temp into proper sections, minimized modules

JeremiahPetersen committed:
moved gpu temp to new bullet under gpu name

JeremiahPetersen committed:
cleaned up imports

JeremiahPetersen commented:
I added GPU temp (updates every second) as a bullet in Graphics
I added CPU temp (updates every second) as a bullet in Processor
GPU temp is working on all tests so far
CPU temp is not updating every second as expected on windows platforms. (mechanism is working, so I am working to find another solution to this today.)

JeremiahPetersen committed:
cleaned up def var names

JeremiahPetersen committed:
added GPU utilization

diamant3 commented:
Ok, let me know if it's ready for review. Thanks!

JeremiahPetersen committed:
cleaned code

JeremiahPetersen commented:
It is ready for review now.  The gpu temp and utilization are working.  The cpu temp is working, but for some windows / intel systems, it would need additional software (Open Hardware Monitor OHM) which would be adding too much and may be unnecessary.  I cleaned up the code.  Let me know what changes you would like.

diamant3 commented on requirements.txt:
```diff
@@ -1,6 +1,7 @@
+humanize
 dearpygui
-py-cpuinfo
+cpuinfo
```
This is py-cpuinfo not cpuinfo

diamant3 commented on sysig.py:
```diff
@@ -110,19 +173,63 @@ def get_cpu_util():
+                        amd_manager = pyadl.ADLManager.getInstance()
+                        devices = amd_manager.getDevices()
+                        for device in devices:
+                            temperature_data = device.getTemperature()
```
I think it's `getCurrentTemperature()`, please check their docs if correct

JeremiahPetersen replied:
I checked here - https://github.com/nicolargo/pyadl - and you are correct.  I made the changes.

diamant3 replied:
Okay, thanks!

diamant3 commented on sysig.py:
```diff
@@ -33,7 +50,52 @@ def get_cpu_util():
+            temperature_info = w.MSAcpi_ThermalZoneTemperature()[0]
+            return temperature_info.CurrentTemperature / 10.0 - 273.15
+        except wmi.x_access_denied:
+            return "Access Denied (Run script as administrator)"
```
Is there any solution for this without using administrator access in Windows? All of the modules I used don't need administrator access.

JeremiahPetersen replied:
That was the problem I was running into, I researched the problem for a while and the solutions were to run wmi using admin access, or use additional software like OHM.  As the graphics additions are working, I can scrap the cpu temp idea until I can figure out a better solution for Windows / Intel.  I can remove the cpu temperature code for now if you would like to incorporate just the gpu additions.

diamant3 replied:
> use additional software like OHM.

Please don't, This is just a simple app 😅

> I can remove the cpu temperature code for now if you would like to incorporate just the gpu additions.

If it's okay with you, Please do. It's better to do GPU additions for now, and you can submit another PR if you have a solution for admin access.

JeremiahPetersen replied:
lol yea I like that it is a simple app.  I was trying to implement an easy way to see cpu temp but it turned out to be a rabbit hole lol.  I like keeping it simple.  That sounds good.  I will edit the code and send that in a moment

diamant3 replied:
Thanks! I will run the workflow and merge this later.

diamant3 reviewed (changes requested):
I've reviewed it all now. I will run the workflow and merge it later if it's running correctly.

JeremiahPetersen committed:
fixed AMD GPU getCurrentTemperature()

JeremiahPetersen committed:
removed cpu temp, cleaned code

JeremiahPetersen commented:
This is good for review, I removed the cpu temp and will continue researching to see if it is possible to do in a simple way.  If not, no worries.  The gpu additions should all be working without issue now.  Let me know if there are any other changes to make.  Thanks for letting me contribute and providing clear direct feedback!

diamant3 commented:
Don't worry it's just errors for variable naming, length of line etc., I'll merge this later. thanks again!

JeremiahPetersen commented:
no worries!  I was going through each of the errors and realizing that lol

diamant3 commented:
Thanks!

JeremiahPetersen commented:
You are welcome!  Thank you!
